### ISSUE TITLE
[bug report] custom theme

### ISSUE THREAD

**Issue by Gintoki-C** [element-plus/element-plus#3855]
### Element Plus version
1.1.0-beta.20

### OS/Browsers version
win10 edge94.0.992.31 (官方内部版本) (64 位)

### Vue version
3.0.0

### Reproduction Link
https://codesandbox.io/s/element-plus-spa-forked-kmw5w?file=/element-variables.scss

### Steps to reproduce
新建element-variables.scss中
找不到字体文件

### What is Expected?
无

### What is actually happening?
These relative modules were not found:

* ./fonts/element-icons.ttf in ./node_modules/css-loader/dist/cjs.js??ref--8-oneOf-3-1!./node_modules/postcss-loader/src??ref--8-oneOf-3-2!./node_modules/sass-loader/dist/cjs.js??ref--8-oneOf-3-3!./src/assets/style/element-variables.scss
* ./fonts/element-icons.woff in ./node_modules/css-loader/dist/cjs.js??ref--8-oneOf-3-1!./node_modules/postcss-loader/src??ref--8-oneOf-3-2!./node_modules/sass-loader/dist/cjs.js??ref--8-oneOf-3-3!./src/assets/style/element-variables.scss

**Comment by Gintoki-C** [comment]
> 我也有这个问题，请问解决了吗？

**按照这样配置**
```
@forward "element-plus/theme-chalk/src/common/var.scss"with ($font-path: '~element-plus/dist/fonts',
    $colors: ("primary": ("base": #8baa41,
        ),
        'success': ('base': #1d953f,
        ),
        'warning': ('base': #ffc20e,
        ),
        'danger': ('base': #f05b72,
        ),
        'error': ('base': #f05b72,
        ),
        'info': ('base': #a1a3a6,
        ),
    ),
);
```

**Comment by Gintoki-C** [comment]
> Hi! I'm having the same problem. Is it fixed? If so, how can I fix it?

Hi!, please try this configuration
https://github.com/element-plus/element-plus/issues/3855#issuecomment-943938817

**Comment by quiet201** [comment]
> > Hi! I'm having the same problem. Is it fixed? If so, how can I fix it?
> 
> Hi!, please try this configuration [#3855 (comment)](https://github.com/element-plus/element-plus/issues/3855#issuecomment-943938817)



> Hi! I'm having the same problem. Is it fixed? If so, how can I fix it?



> 入口文件： import { createApp } from 'vue'; import './assets/styles/elementvari.scss'; import ElementPlus from 'element-plus'; import 'element-plus/dist/index.css'; import App from './App.vue';
> 
> const app = createApp(App); app.use(ElementPlus);
> 
> elementvari.scss文件 @forward "element-plus/theme-chalk/src/common/var.scss"with ($font-path: '~element-plus/dist/fonts', $colors: ("primary": ("base": #8baa41, ), 'success': ('base': #1d953f, ), 'warning': ('base': #ffc20e, ), 'danger': ('base': #f05b72, ), 'error': ('base': #f05b72, ), 'info': ('base': #a1a3a6, ), ), );
> 
> 是这样配置吗？还需要改其他地方吗，我还是没有效果，只是不报错了

下面这个是我的  
**********************************************
main.js 代码：

import { createApp } from 'vue'
import App from './App.vue'
import router from './router'
import ElementPlus from 'element-plus'  
import '@/styles/element-plus.scss'. // 自定义的scss  详细看下面配置 

const app = createApp(App) 
app
  .use(router)
  .use(ElementPlus)
  .mount('#app')

******************************************************************
element-plus.scss 代码:

$colors: (
  'primary': (
    'base': #1890ff,
  ),
  'success': (
    'base': #13ce66,
  ),
  'warning': (
    'base': #ffba00,
  ),
  'danger': (
    'base': #ff4949,
  ),
  'error': (
    'base': #db2828,
  ),
  'info': (
    'base': #42b8dd,
  ),
);
 
@forward "~element-plus/theme-chalk/src/common/var.scss" with (
  // do not use same name, it will override.
  $colors: $colors,
  $font-path: '~element-plus/dist/fonts'
);
 
@import '~element-plus/theme-chalk/src/index.scss';

**Comment by quiet201** [comment]
> 入口文件： import { createApp } from 'vue'; import './assets/styles/elementvari.scss'; import ElementPlus from 'element-plus'; import 'element-plus/dist/index.css'; import App from './App.vue';
> 
> const app = createApp(App); app.use(ElementPlus);
> 
> elementvari.scss文件 @forward "element-plus/theme-chalk/src/common/var.scss"with ($font-path: '~element-plus/dist/fonts', $colors: ("primary": ("base": #8baa41, ), 'success': ('base': #1d953f, ), 'warning': ('base': #ffc20e, ), 'danger': ('base': #f05b72, ), 'error': ('base': #f05b72, ), 'info': ('base': #a1a3a6, ), ), );
> 
> 是这样配置吗？还需要改其他地方吗，我还是没有效果，只是不报错了

你这里把 css和 scss 都引入了  去掉 css引入 看下

**Comment by bellasource** [comment]
按照你的终于解决了 赞

**Comment by mebugs** [comment]
```javascript
// styles/element/index.scss
/* just override what you need */
@forward "element-plus/theme-chalk/src/common/var.scss" with (
  $colors: (
    "primary": (
      "base": green,
    ),
  ),
  $font-path: "element-plus/theme-chalk/src/fonts"
);

@import "element-plus/theme-chalk/src/index.scss";
```

需要增加：$font-path: "element-plus/theme-chalk/src/fonts"，覆盖默认的变量（我的本地环境：Webpack+Vue3.2）
ElE-Plus的版本：1.1.0-beta.24

**Comment by Gintoki-C** [comment]
> ```js
> // styles/element/index.scss
> /* just override what you need */
> @forward "element-plus/theme-chalk/src/common/var.scss" with (
>   $colors: (
>     "primary": (
>       "base": green,
>     ),
>   ),
>   $font-path: "element-plus/theme-chalk/src/fonts"
> );
> 
> @import "element-plus/theme-chalk/src/index.scss";
> ```
> 
> 需要增加：$font-path: "element-plus/theme-chalk/src/fonts"，覆盖默认的变量（我的本地环境：Webpack+Vue3.2） ElE-Plus的版本：1.1.0-beta.24

beta.20版本我加了,运行也可以
```
$font-path: '~element-plus/dist/fonts',
```

**Comment by Gintoki-C** [comment]
> 24版本好像修复了
没有升级，还是用的.20 等更新试试

**Comment by codeabilly** [comment]
This theming issue kept bugging me for some time... and then it hit me...
While compiling scss I was getting stuff like this in my source:
```css
@use './base.scss';
```

Which means my compiler does not support `@use` command.

So I kept digging further and came across this [github issue](https://github.com/sass/node-sass/issues/2886) 

TL;DR if you're using VUE-cli to create your project you should opt out for cherry picking features you want to install and pick carefully when you get to the step for picking your CSS preprocessor.
For me personally it meant going with `Sass/SCSS (with dart-sass)` instead of `Sass/SCSS (with node-sass)`. 
With that subtle change, following block of code works like magic:
```scss
@forward "~element-plus/theme-chalk/src/common/var.scss" with (
  $colors: (
    "primary": (
      "base": red,
    ),
  ),
);

@import "~element-plus/theme-chalk/src/index.scss";
```

PS. I would like to see some warning in the documentation  stating  you should not use `Sass/SCSS (with node-sass)` in order to get theming to work properly.

**Comment by comod** [comment]
It took my a couple of hours to realize that it is at least important to import 
`@import "element-plus/theme-chalk/src/index.scss";`
In the docs this line is out-commented (which means not important; nobody reads that then!)
instead a complicated vite-setup with additional loaders and messy stuff comes up right next which is in my oppionion a "extended/advanced" setup.
Please clean up the docs here a bit. Took too much time again to see where is the Problem :-/

**Comment by Cally99** [comment]
@comod I agree the docs are so messy and the migration from vue2 to vue 3 is so poorly written or non-existent

**Comment by Dsaquel** [comment]
I assume this response https://github.com/element-plus/element-plus/issues/3855#issuecomment-946316636 is the right way to go.

Also there is a good example in this repo in how to extand the theme: https://github.com/element-plus/element-plus-vite-starter/blob/main/src/styles/element/index.scss#L1-L20.

Closing the issue as it is answered by the response above.
